app/src/main/java/com/mrkenhoff/qalculate/ui/ButtonLayoutFragment.kt

```diff
@@ -2,23 +2,30 @@ package com.mrkenhoff.qalculate.ui
 
 import android.content.res.Resources
 import android.os.Bundle
+import android.util.Log
 import android.util.TypedValue
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.LinearLayout
-import androidx.appcompat.view.ContextThemeWrapper
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
 import com.mrkenhoff.qalculate.R
+import dagger.hilt.android.AndroidEntryPoint
 import org.w3c.dom.Element
 import org.w3c.dom.Node
 import javax.xml.parsers.DocumentBuilderFactory
+import org.greenrobot.eventbus.EventBus
 
 
 /**
  * A [Fragment] subclass that shows a button layout.
  */
+@AndroidEntryPoint
 class ButtonLayoutFragment : Fragment() {
+
+    private val viewModel: MainViewModel by activityViewModels()
```

Comment:
Why does this child fragment need direct access to the parent view? That makes it harder to re-use fragments in other contexts.

Reply:
Thanks for spotting that. Was a leftover...